File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def helper(self,root):
        if(root==None): return 1e5
        if(root.left==None and root.right==None): return 1
        left=self.helper(root.left)
        right=self.helper(root.right)
        return min(left,right)+1
    def minDepth(self, root):
        ans=self.helper(root)
        if (ans>=1e5): return 0 
        return ans

File: minimum-depth-of-binary-tree/minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int helper(TreeNode* root){
        if(!root) return 1e5;
        if(!root->left && !root->right) return 1;
        int left=helper(root->left);
        int right=helper(root->right);
        return min(left,right)+1;
    }
    int minDepth(TreeNode* root) {
        if(!root) return 0;
        return helper(root);
    }
};